src/ImageSharp/Image/IImage.cs

```diff
@@ -0,0 +1,31 @@
+﻿using SixLabors.ImageSharp.Formats;
+using SixLabors.ImageSharp.MetaData;
+
+namespace SixLabors.ImageSharp
+{
+    /// <summary>
+    /// Represents the base image abstraction.
+    /// </summary>
+    public interface IImage
+    {
+        /// <summary>
+        /// Gets information about pixel.
+        /// </summary>
+        PixelTypeInfo PixelType { get; }
```

Comment:
I suppose this value holds the native pixel type in the stream, after `Identify()`.
What's the deal with `Image.Load<TPixel>()`? What if `typeof(TPixel)` ≠ `the native pixel type in the stream` ?

Reply:
That situation will happen more often than not since 3 out of 4 of our currently supported image formats are commonly RGB.

If it's going to be part of the interface then both The property and its type should be renamed as it's just confusing now. What would the equivalent information be called in other libraries?

Reply:
However, if we rename it to something like `OriginalPixelType`, while keeping it in `IImage` means that we added a stream-specific metadata-like property to the `Image<T>` class which is mostly intended to represent pixel data.

My suggestion:
Let's name it `OriginalPixelType` and move it into the `ImageMetaData` class.

Reply:
Image<T> shouldn't have any knowledge of its original loaded data.. this is why we removed the original image format from Image<T> I think we need this API to return an `ImageMetadata` object instead that exposes all this info and not the based `IImage` interface.

Reply:
Okkay, let's sum up the situation, so we can figure it out:
- @tocsoft Not sure if I'm comfortable with this fact, but we already [have `MetaData` exposed](https://github.com/SixLabors/ImageSharp/blob/master/src/ImageSharp/Image/Image%7BTPixel%7D.cs#L101) on `Image<TPixel>`.
- We need to include Width+Height (or maybe `SixLabors.Size`?) into the metadata-only result. Currently it's not exposed on `ImageMetadata`.
  - @denisivan0v solved this by defining `ImageInfo : IImage` as the metadata-only result
  - But maybe these are just different things, and sharing a common interface is an LSP violation.

@denisivan0v @xakep139 do you actually need the `ImageInfo`/`Image<T>` polymorphism? Eg: do you want to mix them in heterogenous containers? For me it feels like an unrealistic scenario.

Reply:
>But maybe these are just different things, and sharing a common interface is an LSP violation.

I'm not sure... Reading through the code again I now see why I'm uneasy and suggested the change. It's because it's merging the two identities as one.

See https://github.com/SixLabors/ImageSharp/pull/292/files/9f719d5b9423083c112a69144e98a15bbb8c9409#diff-b4f626cf509fdbafd46cc45da8aed941R429 for example. The code takes the raw input format and passes it into the image, which now contains conflicting information. 

My Suggestion:

- Drop that new overload to the `Image<TPixel>` constructor and calculate the `PixelTypeInfo` property automatically in our current constructor.
- Keep naming as-is, including interface sharing.
- Write decent intellisense documentation describing what `ImageInfo` is, clearly highlighting that it contains the raw info.

That way we preserve two separate identities. The current raw identity that can be used for making informed decisions as to what `TPixel` to use etc, and our point-in-time representation of the image pixels.

Reply:
@JimBobSquarePants I'm not sure if I'm following you correctly. 
Do you suggest to keep the `PixelType` property on the `IImage` interface + keep it in sync with `TPixel` in case of `Image<TPixel>`? Or the opposite: always exposing the raw type after `Image.Load<TPixel>()` regardless of `TPixel`?

Reply:
@antonfirsov The former.

For `Image<TPixel>` the `PixelType` property represents `TPixel`

For `ImageInfo` the `PixelType` property represents bits per pixel of the source image.

Reply:
Makes sense, and looks like being in sync with my vision about the role of `PixelTypeInfo` as bridge between `TPixel` and raw formats, and future pixel-type agnostic Image API-s. (System.Drawing is decoding images into their native formats, we might implement this feature with `IImage`.)

However I'm still unsure if the result of `Identify()` is actually an **image**. Maybe we should rename to `IImageInfo`, and reserve `IImage` for future use cases: 
`IImage` should represent an image that *actually has pixels*, so it could be used with processors, `.Mutate()`, `.Clone()` etc.

I think we should implement the following hierarchy in long term:
```C#
// Has Dimensions + PixelTypeInfo + Metadata
public interface IImageInfo { ... }

// Also has pixels, could be used with ImageProcessor-s.
public interface IImage : IImageInfo { ... }

internal class ImageInfo : IImageInfo { ... }

public class Image<TPixel> : IImage { ... }
```

Reply:
Yeah, let's do the rename. 👍

Reply:
Great design review, thanks!

I also wasn't fully satisfied that `Image<T>` and internal `ImageInfo` shares the same `IImage` interface. But `IImageInfo` and suggested hierarchy solved the problem and made things clear.

Done in 61c9caf.